src/util/util.ts

```diff
@@ -1,75 +1,186 @@
+  interface MPVersionOption extends QuickPickItem {
+    label: string; // label is the long-name that is displaed in vscode
+    version: string; // version is the shortname that is used internally by the microprofile starter api
+  }
+
+  const mpVersionOptions: MPVersionOption[] = [];
+  for (const mpVersion of mpVersions) {
+    // if we have a label defined for the given MP version shortname add it to the options array
+    if (MP_VERSION_LABELS[mpVersion] != null) {
+      mpVersionOptions.push({
+        label: MP_VERSION_LABELS[mpVersion],
+        version: mpVersion,
+      });
     }
+  }
+
+  const mpVersionQuickPickResult = await vscode.window.showQuickPick(mpVersionOptions, {
+    ignoreFocusOut: true,
+    placeHolder: "Select a MicroProfile version.",
+  });
+
+  if (mpVersionQuickPickResult != null) {
+    return mpVersionQuickPickResult.version;
+  }
+
+  return undefined;
 }
 
-export async function mapToPropertiesFile(arrayName : string, originalArray : string[]) {
-	var mappedArray : string[] = new Array();
-	mappedArray =  originalArray.map(function (elem) {
-		return properties[arrayName][elem];
-	});
-	mappedArray.sort().reverse();
-	return mappedArray;
+export async function askForMPserver(mpServers: string[]): Promise<string | undefined> {
+  interface MPServerOption extends QuickPickItem {
+    label: string; // label is the long-name that is displaed in vscode
+    server: string; // server is the shortname that is used internally by the microprofile starter api
+  }
+
+  const mpServerOptions: MPServerOption[] = [];
+  for (const mpServer of mpServers) {
+    // if we have a server label for the given mp server shortname add it to the options array
+    if (MP_SERVER_LABELS[mpServer] != null) {
+      mpServerOptions.push({
+        label: MP_SERVER_LABELS[mpServer],
+        server: mpServer,
+      });
+    }
+  }
+
+  const mpVersionQuickPickResult = await vscode.window.showQuickPick(mpServerOptions, {
+    ignoreFocusOut: true,
+    placeHolder: "Select a MicroProfile server.",
+  });
+
+  if (mpVersionQuickPickResult != null) {
+    return mpVersionQuickPickResult.server;
+  }
+  return undefined;
 }
 
-function capitalizeFirstLetter(newString :string) {
-    return newString.charAt(0).toUpperCase() + newString.slice(1);
+export async function askForMPSpecifications(
+  specs: string[],
+  specDescriptions: Record<string, string>
+): Promise<Array<string> | undefined> {
+  interface MPSpecOption extends QuickPickItem {
+    spec: string; // spec is the shortname used internally my microprofile starter api
+    label: string; // name of mp spec
+    detail: string; // description of mp spec
+  }
+
+  const mpSpecOptions: MPSpecOption[] = specs.map(spec => {
+    const fullDescription = specDescriptions[spec];
+    const [name, desc] = fullDescription.split("-");
+    return {
+      spec: spec,
+      label: name,
+      detail: desc,
```

Comment:
can change this to `detail: capitalizeFirstLetter(desc)` now that you have this method implemented.